web-admin/src/features/projects/status/ProjectResources.svelte

```diff
@@ -17,15 +29,60 @@
               resource.meta.name.kind !== "rill.runtime.v1.ProjectParser",
           );
         },
+        refetchOnMount: true,
+        refetchInterval: isReconciling ? 500 : false,
       },
     },
   );
+
+  $: isAnySourceOrModelReconciling = $resources?.data?.some(
+    (resource) =>
+      resource.meta.reconcileStatus ===
+        V1ReconcileStatus.RECONCILE_STATUS_PENDING ||
+      resource.meta.reconcileStatus ===
+        V1ReconcileStatus.RECONCILE_STATUS_RUNNING,
+  );
+
+  function refreshAllSourcesAndModels() {
+    isReconciling = true;
+
+    void $createTrigger.mutateAsync({
+      instanceId: $runtime.instanceId,
+      data: {
+        allSourcesModelsFull: true,
+      },
+    });
+
+    void queryClient.invalidateQueries(
+      getRuntimeServiceListResourcesQueryKey(
+        $runtime.instanceId,
+        // All resource "kinds"
+        undefined,
+      ),
+    );
+  }
+
+  $: if (!isAnySourceOrModelReconciling) {
+    isReconciling = false;
+  }
 </script>
 
 <section class="flex flex-col gap-y-4">
-  <h2 class="text-lg font-medium">Resources</h2>
-  {#if $resources.isLoading}
-    <DelayedSpinner isLoading={$resources.isLoading} size="16px" />
+  <div class="flex items-center justify-between">
+    <h2 class="text-lg font-medium">Resources</h2>
+    <Button type="secondary" on:click={refreshAllSourcesAndModels}>
+      {#if isReconciling}
+        Refreshing...
+      {:else}
+        Refresh all sources and models
+      {/if}
+    </Button>
+  </div>
+  {#if $resources.isLoading || $resources.isRefetching}
```

Comment:
This check for `isRefetching` causes a flicker. [See Jam](https://jam.dev/c/5b356cc8-b650-4145-bca9-8af997c7ce21)